update-center-update/uc-remediation.groovy

```diff
@@ -0,0 +1,334 @@
+// methods below
+// ----------------------------------------------------------------------------------------------------
+
+/**
+  * removes the uc-remediation.groovy script from the filesystem
+  */
+def removeScript() {
+    File f = new File(Jenkins.getInstance().getRootDir().getAbsolutePath() + "/init.groovy.d/uc-remediation.groovy");
+    if (f.exists()) {
+        debug("Removing script " + f.getAbsolutePath());
+        if (!_dry_run) {
+            f.delete();
+            info("Script has been uninstalled");
+        } else {
+            info("dry run, not removing script file");
+        }
+        return true;
+    } else {
+        info(f.getAbsolutePath() + " not found, skipping uninstall");
+        return false;
+    }
+}
+
+def isCertificateCheckingEnabled() {
+	debug("DownlaodService.signatureCheck == " + DownloadService.signatureCheck);
+    return DownloadService.signatureCheck;
+}
+
+def disableCertificateValidation() {
+    info("disabling certificate validation");
+    setCertificateValidation(false);
+}
+
+def enableCertificateValidation() {
+    info("enabling certificate validation");
+    setCertificateValidation(true);
+}
+
+def setCertificateValidation(boolean check) {
+    debug("DownloadService.signatureCheck original value [" + DownloadService.signatureCheck + "]");
+    if (!_dry_run) {
+        DownloadService.signatureCheck = check;
+        debug("DownloadService.signatureCheck new value [" + DownloadService.signatureCheck + "]");
+    } else {
+        info("dry run, not changing signature check");
+    }
+}
+
+// this currently makes the assumption that any error checking the online uc means we are airgapped
+// possibly this needs to be more fine-grained?
+// returns false if the online uc is able to be validated, true otherwise
+def isAirGapped() {
+    return(!checkOnlineUC(false));
+}
+
+def checkOfflineUC(boolean validate) {
+    return checkUpdateSite(getDefaultOfflineUC(), validate);
+}
+
+def checkOfflineUC() {
+    return checkUpdateSite(getDefaultOfflineUC(), DownloadService.signatureCheck);
+}
+
+def checkOnlineUC(boolean validate) {
+    return checkUpdateSite(getDefaultOnlineUc(), validate);
+}
+
+def checkOnlineUC() {
+    return checkUpdateSite(getDefaultOnlineUc(), DownloadService.signatureCheck);
+}
+
+def checkUpdateSite(UpdateSite site, boolean validate) {
+    // ugly to accomodate the logic in nectar-license plugin
+    originalCheckValue = isCertificateCheckingEnabled();
+    if (validate && !isCertificateCheckingEnabled()) {
+        setCertificateValidation(true);
+    }
+
+    try {
+        debug("checking update site " + site + " with validate == " + validate);
+        FormValidation v = site.updateDirectlyNow(validate);
+        debug("form validation -> "  + v);
+        if (v.kind == FormValidation.Kind.OK) {
+            debug(site.getUrl() + " is ok");
+            setCertificateValidation(originalCheckValue);

```

Comment:
If Groovy is following the same rules as Java, this could go in a `finally` block. It would be executed even after the return.